backend/endpoints/threads.py

```diff
@@ -164,96 +158,83 @@ async def get_thread_messages(org_id: str, room_id: str, message_id: str):
 ):
 
-    payload = request.dict()
-    loadedMessage = await get_message(org_id, room_id, message_id)
-    loadedMessageThread = loadedMessage['threads']
-    thread_to_add = {"threads": [payload, *loadedMessageThread]}
+    """Updates a thread message in a thread in a room.
 
-    if not loadedMessage:
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND, detail="Message not found"
-        )
+    Edits a thread in an existing message document in the messages database collection
+    while publishing to all members of the room in the background.
 
-    added_thread_message = await edit_message_threads(org_id, message_id, thread_to_add)
+    Args:
+        org_id: A unique identifier of the organization.
+        room_id: A unique identifier of the room.
+        message_id: A unique identifier of the message that is being edited.
+        thread_id: A unique identifier of the thread that is being edited.
+        request: A pydantic schema that defines the message request parameters.
+        background_tasks: A background task for publishing to all
+                          members of the room.
 
-    if not added_thread_message:
-        raise HTTPException(
-            status_code=status.HTTP_424_FAILED_DEPENDENCY,
-            detail={"thread not added": added_thread_message},
-        )
+    Returns:
+        A dict containing data about the thread that was edited.
 
-    # payload["edited"] = True
-    loadedMessage.update(thread_to_add)
+            {
+                "status": "success",
+                "message": "Thread Updated",
+                "data": {
+                    "sender_id": "63cb53b42009ec8a16a5774b",
+                    "emojis": [],
+                    "richUiData": {
+                    "blocks": [
+                        {
+                        "data": {},
+                        "depth": 0,
+                        "entityRanges": [],
+                        "inlineStyleRanges": [],
+                        "key": "eljik",
+                        "text": "Steady green like what??",
+                        "type": "unstyled"
+                        }
+                    ]
+                    },
+                    "files": [],
+                    "saved_by": [],
+                    "timestamp": 0,
+                    "created_at": "2023-01-31 08:53:04.126997",
+                    "edited": true,
+                    "thread_id": "b39cfddc-a0a7-11ed-b15b-b8819887ed7a"
+                }
+            }
+    Raises:
+        HTTPException [401]: You are not authorized to edit this thread message.
+        HTTPException [404]: Message not found.
+        HTTPException [424]: thread not updated.
+    """
 
-    # Publish to centrifugo in the background.
-    background_tasks.add_task(
-        centrifugo_client.publish,
-        room_id,
-        Events.MESSAGE_UPDATE,
-        loadedMessage
-    )
+    message = await get_message(org_id, room_id, message_id)
```

Comment:
why do we need to get the message here? we are making another db call in `threads_utils` to get the same message.

Reply:
oh right, I could just return the message from the db call we made in `threads_utils` instead then